pkg/codegen/pcl/binder_schema.go

```diff
@@ -549,6 +549,7 @@ func GenEnum(
 	from model.Expression,
 	safeEnum func(member *schema.Enum),
 	unsafeEnum func(from model.Expression),
+	diags *hcl.Diagnostics,
```

Comment:
I'm curious why you chose to pass `diags` as an argument instead of returning the `*hcl.Diagnostic`. It feels like a very C like api for go.